services/QuillLMS/client/app/bundles/Staff/components/comprehension/rulesAnalysis/ruleAnalysis.tsx

```diff
@@ -0,0 +1,231 @@
+import * as React from "react";
+import { queryCache, useQuery } from 'react-query';
+import { Link } from 'react-router-dom';
+import stripHtml from "string-strip-html";
+import moment from 'moment';
+import ReactTable from 'react-table';
+
+import { fetchRule } from '../../../utils/comprehension/ruleAPIs';
+import { fetchActivity } from '../../../utils/comprehension/activityAPIs';
+import { fetchRuleFeedbackHistoriesByRule } from '../../../utils/comprehension/ruleFeedbackHistoryAPIs';
+import { fetchConcepts, } from '../../../utils/comprehension/conceptAPIs';
+import { createOrUpdateFeedbackHistoryRating } from '../../../utils/comprehension/feedbackHistoryRatingAPIs';
+import { DataTable, Error, Spinner } from '../../../../Shared/index';
+
+const ALL = 'All'
+const SCORED = 'Scored'
+const UNSCORED = 'Unscored'
+
+const RuleAnalysis = ({ history, match }) => {
+  const { params } = match;
+  const { activityId, ruleId } = params;
+
+  const [filter, setFilter] = React.useState(ALL)
+
+  const { data: conceptsData } = useQuery({
+    queryKey: ['concepts', ruleId],
+    queryFn: fetchConcepts
+  });
+
+  const { data: ruleData } = useQuery({
+    queryKey: [`rule-${ruleId}`, ruleId],
+    queryFn: fetchRule
+  });
+
+  const { data: activityData } = useQuery({
+    queryKey: [`activity-${activityId}`, activityId],
+    queryFn: fetchActivity
+  });
+
+  const { data: ruleFeedbackHistoryData } = useQuery({
+    queryKey: [`rule-feedback-histories-by-rule-${ruleId}`, ruleId],
+    queryFn: fetchRuleFeedbackHistoriesByRule
+  })
+
+  function handleFilterChange(e) { setFilter(e.target.value) }
+
+  function filterResponses(r) {
+    if (filter === ALL) { return true }
+    if (filter === SCORED && r.strength !== null) { return true }
+    if (filter === UNSCORED && r.strength === null) { return true }
+
+    return false
+  }
+
+   async function makeStrong(response) { updateFeedbackHistoryStrength(response.response_id, true) }
```

Comment:
what about naming it updateFeedbackHistoryRatingStrength, which is what it is accomplishing?